platforms/scripts/afu_synth_setup_std

```diff
@@ -0,0 +1,251 @@
+        raise
+
+    # Make target "hw" directory
+    os.mkdir(os.path.join(dst, 'hw'))
+
+    # Make a scripts directory with clean.sh and run.sh
+    scripts_src = os.path.abspath(os.path.join(base_src, 'bin'))
+    scripts_dst = os.path.join(dst, 'scripts')
+    os.mkdir(scripts_dst)
+    os.symlink(os.path.join(scripts_src, 'clean.sh'),
+               os.path.join(scripts_dst, 'clean.sh'))
+    run_sh = os.path.join(scripts_dst, 'run.sh')
+    with open(run_sh, 'w') as f:
+        f.write('#!/bin/sh\n')
+        f.write('export BBS_LIB_PATH=${BBS_LIB_PATH:-"' +
+                os.path.join(base_src, 'hw/lib') + '"}\n')
+        f.write(os.path.join(scripts_src, 'run.sh') + ' $*\n')
+    st = os.stat(run_sh)
+    os.chmod(run_sh, st.st_mode | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH)
+
+
+#
+# Configure the build environment
+#
+def setup_build(args):
+    # Does the sources file exist?
+    if (not os.path.exists(args.sources)):
+        errorExit("Sources file {0} not found.".format(args.sources))
+
+    # Transform path to sources relative to build directory
+    is_abs = os.path.isabs(args.sources)
+    if (is_abs):
+        sources = args.sources
+    else:
+        sources = os.path.relpath(args.sources,
+                                  os.path.join(args.dst, 'build'))
+
+    # Get the JSON file
+    os.chdir(os.path.join(args.dst, 'hw'))
+    try:
+        cmd = ['rtl_src_config']
+        if (is_abs):
+            cmd.append('--abs')
+        cmd.append('--json')
+        cmd.append(sources)
+        json = commands_list_getoutput(cmd)
+    except Exception:
+        errorExit("Failed to read sources from {0}".format(sources))
+
+    json = json.strip().split('\n')
+    if (len(json) == 0):
+        errorExit("No AFU JSON file found in {0}".format(sources))
+    if (len(json) > 1):
+        errorExit("More than one AFU JSON file found in {0}".format(sources))
+
+    json = json[0]
+    os.symlink(json, os.path.basename(json))
+
+    # Configure sources
+    print('Changing to directory ' + os.path.join(args.dst, 'build'))
+    os.chdir('../build')
+    cmd = ['rtl_src_config']
+    if (is_abs):
+        cmd.append('--abs')
+    cmd.append('--qsf')
+    cmd.append(sources)
+    os.system(' '.join(cmd) + ' > ../hw/afu.qsf')
+
+    # Configure the platform
+    cmd = 'afu_platform_config --qsf --tgt platform'
+    cmd += ' --src=' + json + ' '
+
+    # Was the platform specified on the commadn line?
+    if (args.plat):
+        cmd += args.plat
+    else:
+        # Get the platform from the release
+        plat_class_file = os.path.join(getSysReleasePath(args),
+                                       'hw/lib/fme-platform-class.txt')
+        with open(plat_class_file) as f:
+            cmd += f.read().strip()
+
+    os.system(cmd)
+
+    # Extract JSON info for Verilog
+    cmd = 'afu_json_mgr json-info --afu-json=' + json
+    cmd += ' --verilog-hdr=../hw/afu_json_info.vh'
+    os.system(cmd)
+
+
+def main():
```

Comment:
Please verify my understanding of the end user flow. The user creates the AFU source specification file, points that to this top-level script along with release and platform info. This script generates the QII project and clean/run.sh scripts (used by the current flow?). What is the main value here? Ability to centrally manage AFU project settings for a particular release?

Can you clarify the term "platform"? Are variants of DCP (inspur, rushcreek) considered separate platforms?

Reply:
Platform is a physical platform for which there is a unique release. Inspur, Rush Creek, BDX Xeon+FPGA, etc.

This script is the analog of afu_sim_setup. It reads the same source specification file but, instead of constructing an ASE environment, constructs a Quartus build environment. Once this is checked in I will check in the script that transforms existing release to work with this script and the rest of the platform interface manager. The changes are pretty minor, especially for DCP, since I took Chip's structure as a starting point and rearranged the old Xeon+FPGA releases to match.

Reply:
Got it. Thanks.